x-pack/plugins/observability_solution/apm/server/routes/assistant_functions/get_log_rate_analysis/index.ts

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { ElasticsearchClient } from '@kbn/core-elasticsearch-server';
+import type { CoreRequestHandlerContext } from '@kbn/core/server';
+import { aiAssistantLogsIndexPattern } from '@kbn/observability-ai-assistant-plugin/server';
+import { fetchLogRateAnalysis } from '@kbn/aiops-log-rate-analysis/queries/fetch_log_rate_analysis';
+import { APMEventClient } from '../../../lib/helpers/create_es_client/create_apm_event_client';
+
+export async function getLogRateAnalysis({
+  apmEventClient,
+  esClient,
+  coreContext,
+  arguments: args,
+}: {
+  apmEventClient: APMEventClient;
+  esClient: ElasticsearchClient;
+  coreContext: Pick<CoreRequestHandlerContext, 'uiSettings'>;
+  arguments: {
+    start: string;
+    end: string;
+    entities: {
+      'service.name'?: string;
+      'host.name'?: string;
+      'container.id'?: string;
+      'kubernetes.pod.name'?: string;
+    };
+  };
+}): ReturnType<typeof fetchLogRateAnalysis> {
+  const index = await coreContext.uiSettings.client.get<string>(aiAssistantLogsIndexPattern);
+  return await fetchLogRateAnalysis({
+    esClient,
+    arguments: {
+      index,
+      start: args.start,
+      end: args.end,
+      timefield: '@timestamp',
+    },
+  });
```

Comment:
The response from `fetchLogRateAnalysis` is currently sent untransformed to the LLM. I suggest we remove any data that we don't expect the LLM to be able to understand to reduce noise. `significantItems` seems like the most important. `windowParameters` and `dateHistogramBuckets` on the other hand doesn't seem terribly important and we may want to omit them.
WDYT @dgieselaar ?


```json
{
  "key": "logRateAnalysis",
  "description": "Significant field/value pairs in log data that contributed to changes in the log rate.",
  "data": {
    "logRateChange": {
      "type": "spike",
      "timestamp": 1719409656000,
      "logRateChangeCount": 15581,
      "averageLogRateCount": 5795,
      "logRateAggregationIntervalUsedForAnalysis": "a few seconds",
      "documentSamplingFactorForAnalysis": 0.1,
      "extendedChangePoint": {
        "startTs": 1719409464000,
        "endTs": 1719410040000
      }
    },
    "significantItems": [],
    "dateHistogramBuckets": {
      "1719408840000": 5473,
      "1719408864000": 13756,
      "1719408888000": 14171,
      "1719408912000": 14903,
      "1719408936000": 15225,
      "1719408960000": 13459,
      "1719408984000": 13005,
      "1719409008000": 1269,
      "1719409032000": 395,
      "1719409056000": 284,
      "1719409080000": 391,
      "1719409104000": 376,
      "1719409128000": 335,
      "1719409152000": 357,
      "1719409176000": 321,
      "1719409200000": 282,
      "1719409224000": 381,
      "1719409248000": 305,
      "1719409272000": 371,
      "1719409296000": 268,
      "1719409320000": 314,
      "1719409344000": 344,
      "1719409368000": 234,
      "1719409392000": 0,
      "1719409416000": 0,
      "1719409440000": 0,
      "1719409464000": 8495,
      "1719409488000": 13564,
      "1719409512000": 13804,
      "1719409536000": 14125,
      "1719409560000": 13367,
      "1719409584000": 13748,
      "1719409608000": 14173,
      "1719409632000": 14410,
      "1719409656000": 15581,
      "1719409680000": 14873,
      "1719409704000": 14491,
      "1719409728000": 14086,
      "1719409752000": 12799,
      "1719409776000": 15265,
      "1719409800000": 15139,
      "1719409824000": 14291,
      "1719409848000": 14002,
      "1719409872000": 15321,
      "1719409896000": 14048,
      "1719409920000": 14248,
      "1719409944000": 14656,
      "1719409968000": 13425,
      "1719409992000": 13673,
      "1719410016000": 13690,
      "1719410040000": 8707,
      "1719410064000": 189,
      "1719410088000": 0,
      "1719410112000": 0,
      "1719410136000": 0,
      "1719410160000": 0,
      "1719410184000": 0,
      "1719410208000": 0,
      "1719410232000": 0,
      "1719410256000": 0,
      "1719410280000": 0,
      "1719410304000": 0,
      "1719410328000": 0,
      "1719410352000": 0,
      "1719410376000": 0,
      "1719410400000": 0,
      "1719410424000": 0,
      "1719410448000": 0,
      "1719410472000": 0,
      "1719410496000": 0,
      "1719410520000": 0,
      "1719410544000": 0,
      "1719410568000": 0,
      "1719410592000": 0,
      "1719410616000": 0,
      "1719410640000": 0
    },
    "windowParameters": {
      "baselineMin": 1719407664000,
      "baselineMax": 1719409464000,
      "deviationMin": 1719409488000,
      "deviationMax": 1719410040000
    }
  }
}

```